### PR TITLE
Throw exception when opening an already opened graph on the same thread.

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-graph/src/main/java/ai/grakn/factory/GraknGraphFactoryImpl.java [hunk 1]
```diff
@@ -124,14 +124,14 @@ private void checkClosure(int numOpenTransactions, Supplier<GraknGraph> graphSup
 
     @Override
     public int openGraphTxs() {
-        if(!graphOpen) return 0;
-        return ((AbstractGraknGraph)getGraph()).numOpenTx();
+        if(graph == null) return 0;
+        return ((AbstractGraknGraph) graph).numOpenTx();
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/factory/GraknGraphFactoryImpl.java]: Will this always be 1 now?

File: grakn-test/src/test/java/ai/grakn/test/migration/export/GraphWriterMainTest.java [hunk 2]
```diff
@@ -33,20 +32,27 @@
...
     @BeforeClass
     public static void loadMovieGraph() {
         graph =  engineContext.factoryWithNewKeyspace().getGraph();
     }
 
+    @Before
+    public void setup(){
+        keyspace = graph.getKeyspace();
+        graph.close();
+    }
```
Comment (aelred) [inline grakn-test/src/test/java/ai/grakn/test/migration/export/GraphWriterMainTest.java]: This looks weird to me: we open the graph only once (with `@BeforeClass`), but then we close that same graph before every test?

Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/migration/export/GraphWriterMainTest.java]: This is to get a hold of a generated keyspace.

File: grakn-graph/src/main/java/ai/grakn/factory/GraknGraphFactoryImpl.java (hunk 1 above)
Comment (fppt) [inline grakn-graph/src/main/java/ai/grakn/factory/GraknGraphFactoryImpl.java]: No

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.GraknEngineFailoverIT.whenSubmittingTasksToOneEngineAndRandomlyKillingTheOthers_TheyComplete

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.GraknEngineFailoverIT.whenSubmittingTasksToOneEngineAndRandomlyKillingTheOthers_TheyComplete

**Comment by fppt** [comment]
@aelred could I get another review on this please?

**Pull request merged**